tiny_dnn/activations/tanh_p1m2_layer.h

```diff
@@ -0,0 +1,38 @@
+/*
+    Copyright (c) 2017, Taiga Nomi, Karan Desai
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/layers/layer.h"
+
+namespace tiny_dnn {
+
+class tanh_p1m2_layer : public activation_layer {
+ public:
+  using activation_layer::activation_layer;
+
+  std::string layer_type() const override { return "tanh-scaled-activation"; }
+
+  void forward_activation(const vec_t &x, vec_t &y) {
+    float_t ep;
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      ep   = std::exp(x[j]);
+      y[j] = ep / (ep + (float_t(1) / ep));
+    }
+  }
+
+  void backward_activation(const vec_t &x,
+                           const vec_t &y,
+                           vec_t &dx,
+                           const vec_t &dy) {
+    for (serial_size_t j = 0; j < x.size(); j++) {
```

Comment:
@bery I named it `j` just because it iterates over a `vec_t` and the outer loop which iterates over `tensor_t` in `forward_propagation` or `backward_propagation` have a counter named `i`. It won't cause any errors, but I thouht it might be just better this way. Does it help though ? If not much, then I'll simply use `i`.